battery_level module: fix converting seconds to H:M:S format

hermandavid opened the pull request:
This PR fixes computing of the battery remaining time. The `h` value was not ensured to be an integer which caused issues in function `_hms_to_seconds` trying to call someting like `int("0.0")` and resulting with an error.

lasers reviewed (commented):
Commit history indicated we might not want `:02d` for hours.

hermandavid commented:
I'm sorry for looking into the reasoning behind that change more carefully.

However, this issue still needs to be addressed. When I set `battery_id` to `"all"`, I'm getting the following error

```
invalid literal for int() with base 10: '1.0'.
Traceback
ValueError: invalid literal for int() with base 10: '1.0'
  File "/home/XYZ/desktop/py3status/py3status/module.py", line 948, in run
    response = method()
  File "/home/XYZ/desktop/py3status/py3status/modules/battery_level.py", line 183, in battery_level
    self._refresh_battery_info(battery_list)
  File "/home/XYZ/desktop/py3status/py3status/modules/battery_level.py", line 403, in _refresh_battery_info
    time_remaining_seconds = self._hms_to_seconds(
  File "/home/XYZ/desktop/py3status/py3status/modules/battery_level.py", line 353, in _hms_to_seconds
    h, m, s = [int(i) for i in t.split(":")]
  File "/home/XYZ/desktop/py3status/py3status/modules/battery_level.py", line 353, in <listcomp>
    h, m, s = [int(i) for i in t.split(":")]
```

And it is caused due to improperly formatted output from `_seconds_to_hms`, which in my case looks like 
```
1.0:10:49
```

lasers commented:
I'm not on a laptop so I can't test this easily. I just looked at commit history to see the recent diffs.

I meant that adding `int()` is probably fine. Just without the `:2d` part. See https://github.com/ultrabug/py3status/commit/db9a00cfe442596394576565c50887ac46ef186c or we get `01:10:49`.

hermandavid commented:
May I ask why is it **not** desired to use the `:02d`? I mean, is this for a practical reason or is this just a personal preference? Having output of a fixed length brings consistency and is IMHO much more readable.

I would understand **not** having zero-padded _value_ on days, for example.

lasers commented:
Tbh, I don't care either way. It is just consistency with how it was before the f-string exception.

Also, some people doesn't have battery life that goes beyond 9 hours so they _might_ want to avoid seeing leading zero on hours... such as @ultrabug since he didn't add `:02d` too.

Either way, the fix is good. :+1:

eumiro commented:
Sorry, that was my f-string refactoring and now I saw on my laptop the wrong formatting and came here with an idea, but saw this PR (thanks, @hermandavid!) . Let's build beautiful f-strings and let's avoid those leading zeros in the hours field: 

```
    def _seconds_to_hms(self, secs):
        m, s = divmod(int(secs), 60)
        h, m = divmod(m, 60)
        return f"{h:2}:{m:02}:{s:02}"
```

hermandavid commented:
I have comitted the change to remove the leading zeroes from the hours field.

ultrabug commented:
@eumiro is the proposed change good for you too?

eumiro commented:
@ultrabug it is certainly not wrong and although my comment above may feel a little bit more natural to me (less converting between int and float), I am very happy @hermandavid came up with a fix for a bug that I introduced, so I'd accept it.

ultrabug commented:
Thanks everyone!
